### PR TITLE
Add pg_stat_wal, pg_stat_replication_slots, and new pg_stat_database fields

### LINKED ISSUE #4: PG14 Support

### PR DESCRIPTION
**Pull request by df7cb**
Close #4.

### CODE REVIEW

File: collector/models/pgStatDatabase.go [hunk 1]
```diff
@@ -6,24 +6,33 @@ import (
...
+	TempBytes    int64        `pg:"temp_bytes" help:"Total amount of data written to temporary files by queries in this database"  metric:"temp_bytes_total"`
+	Deadlocks    int64        `pg:"deadlocks" help:"Number of deadlocks detected in this database"  metric:"deadlocks_total"`
+	checksumFailures int64    `pg:"checksum_failures" help:"Number of data page checksum failures detected in this database" metric:"checksum_failures_count"` // new in PG12
+	checksumLastFailure time.Time `pg:"checksum_last_failure" help:"Time at which the last data page checksum failure was detected in this database" metric:"checksum_last_failure"` // new in PG12
+	BlkReadTime  Milliseconds `pg:"blk_read_time" help:"Time spent reading data file blocks by backends in this database" metric:"blk_read_seconds_total"`
+	BlkWriteTime Milliseconds `pg:"blk_write_time" help:"Time spent writing data file blocks by backends in this database" metric:"blk_write_seconds_total"`
+	sessionTime  Milliseconds `pg:"session_time" help:"Time spent by database sessions in this database, in milliseconds" metric:"session_time_total"` // new in PG14
```
Comment (tbe) [inline collector/models/pgStatDatabase.go:30]: Fields must be exported (upercase), so the database library can see and populate them.

File: collector/models/pgStatDatabase.go [hunk 2]
```diff
@@ -6,24 +6,33 @@ import (
...
+	TupInserted  int64        `pg:"tup_inserted" help:"Number of rows inserted by queries in this database" metric:"rows_inserted_total"`
+	TupUpdated   int64        `pg:"tup_updated" help:"Number of rows updated by queries in this database" metric:"rows_updated_total"`
+	TupDeleted   int64        `pg:"tup_deleted" help:"Number of rows deleted by queries in this database" metric:"rows_deleted_total"`
+	Conflicts    int64        `pg:"conflicts" help:"Number of queries canceled due to conflicts with recovery in this database"  metric:"conflicts_total"`
+	TempFiles    int64        `pg:"temp_files" help:"Number of temporary files created by queries in this database"  metric:"temp_files_total"`
+	TempBytes    int64        `pg:"temp_bytes" help:"Total amount of data written to temporary files by queries in this database"  metric:"temp_bytes_total"`
+	Deadlocks    int64        `pg:"deadlocks" help:"Number of deadlocks detected in this database"  metric:"deadlocks_total"`
+	checksumFailures int64    `pg:"checksum_failures" help:"Number of data page checksum failures detected in this database" metric:"checksum_failures_count"` // new in PG12
```
Comment (tbe) [inline collector/models/pgStatDatabase.go:26]: New fields, that are not present in all database versions must be nullable. Please use `sql.NullInt64`, `NullMilliseconds`, ...

**Review by tbe — CHANGES_REQUESTED** [review]
See inline comments. Also, for version dependent fields, the SQL Query must be written by hand. So for example https://github.com/1and1/pg-exporter/blob/3c62cc78463883b712d1e67db222d1e6f75757ee/collector/activity.go#L88

Else, we will get SQL Errors, as the ORM queries all columns per default.

**Review by tbe — APPROVED** [review]

**Comment by tbe** [comment]
Looks good, merged to master

**Comment by df7cb** [comment]
Thanks, but I wasn't finished yet :D. Expect a few more changes to come.

**Pull request merged**